src/modality-agnostic-files/provenance.md

```diff
@@ -0,0 +1,485 @@
+
+### Principles for encoding provenance in BIDS
+
+-   Provenance information SHOULD be included in a BIDS dataset when possible.
+-   If provenance records are included, these MUST be described using the conventions detailed by this specification.
+-   Provenance records MAY be used to reflect the provenance of a full dataset and/or of specific files at any level of the BIDS hierarchy.
+-   Provenance information SHOULD be de-identified as necessary.
+
+### Definitions
+
+**Provenance record**: provenance metadata consists in sets of records. These provenance records can be of 4 types:
+
+-   `Activity`: Activities represent the transformations that have been applied to the data.
+-   `Entity`: Entities represent input or output data for Activities.
+-   `Software`: The Software record describes a software package.
+-   `Environment`: Environments specify the software environment in which the provenance record was obtained.
+
+Provenance records are described as JSON objects in BIDS. This is detailed in the [Provenance files section](#provenance-files).
+
+!!! example
+    The following graph represents examples of links between provenance records. In this example, `Entities` *sub-001_brainmask.nii* and *sub-001_T1w.nii* represent files used by `Activity` *Brain extraction* to generate another file represented by `Entity` *sub-001_T1w_preproc.nii*. `Activities` *Brain extraction* and *Move to MNI* were associated with the `Software` *FSL* and used `Environment` *Linux* as a software environment.
+    ![](../images/provenance_definitions_graph.png)
+
+**Provenance group**: refers to a collection of provenance records corresponding to a consistent set of processings. Defining multiple provenance groups is appropriate when separate sets of processings have been performed on data, and that they differ in purpose or chronology.
+
+## Provenance in sidecar JSON files
+
+Provenance metadata CAN be stored inside the sidecar JSON of any BIDS file (or BIDS-Derivatives file) it applies to.
+This metadata only describes the provenance of the BIDS file.
+
+Any sidecar JSON file CAN include the following keys:
+
+<!-- This block generates a metadata table.
+The definitions of these fields can be found in
+  src/schema/objects/metadata.yaml
+and a guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_metadata_table(
+   {
+      "GeneratedById": "OPTIONAL",
+      "SidecarGeneratedBy": "OPTIONAL",
+      "Digest": "OPTIONAL",
+      "EntityType": "OPTIONAL"
+   }
+) }}
+
+No other field is allowed to describe provenance inside sidecar JSON files.
```

Comment:
ATM sidecar is "open" for defining any other field and some software might...   hence I would just remove this

```suggestion
```

related
- https://github.com/bids-standard/bids-specification/issues/637
- https://github.com/bids-standard/bids-2-devel/issues/80